geometry-builder.cpp

```diff
@@ -760,8 +759,75 @@ void geometry_builder::set_exclude_broken_polygon(int exclude)
     excludepoly = exclude;
 }
 
+void geometry_builder::set_reprojection(struct reprojection *r)
+{
+    reprojection = r;
+}
+
+/** 
+ * Computes area of given polygonal geometry.
+ * Returns 
+ * - the standard GEOS getArea result if area reprojection has not been enabled
+ * - the GEOS getArea resolt of this geometry projected to EPSG:3857 if reprojection has been enabled
+ */
+double geometry_builder::getArea(const geos::geom::Geometry *geom) const
+{
+    // reprojection is not necessary, or has not been asked for.
+    if (!reprojection)
+    {
+        return geom->getArea();
+    }
+
+    // MultiPolygon - return sum of individual areas
+    if (const geos::geom::MultiPolygon* multi = dynamic_cast<const geos::geom::MultiPolygon *>(geom))
+    {
+        double area = 0.0;
+        for (std::size_t i=0; i<multi->getNumGeometries(); i++)
+        {
+            area += getArea(multi->getGeometryN(i));
+        }
+        return area;
+    }
+
+    const geos::geom::Polygon *poly = dynamic_cast<const geos::geom::Polygon *>(geom);
+    if (!poly) return 0.0;
+
+    // standard polygon - reproject, then compute area of outer - areas of holes
```

Comment:
Why compute the area by doing hole subtraction instead of [getArea from geos::geom::Polygon](http://geos.osgeo.org/doxygen/classgeos_1_1geom_1_1Polygon.html#a7fdaf9550f08f56340b0ce8350501b66)